Mathlib/RingTheory/Finiteness/Ideal.lean

```diff
@@ -88,4 +89,9 @@ lemma exists_pow_le_of_le_radical_of_fg {R : Type*} [CommSemiring R] {I J : Idea
     use n₁ + n₂
     exact Ideal.sup_pow_add_le_pow_sup_pow.trans (sup_le hn₁ hn₂)
 
+lemma isNilpotent_iff_le_nilradical {R : Type*} [CommSemiring R] {I : Ideal R} (hI : I.FG) :
+  IsNilpotent I ↔ I ≤ nilradical R :=
+⟨fun ⟨n, hn⟩ _ hx => ⟨n, hn ▸ Ideal.pow_mem_pow hx n⟩,
+  fun h => let ⟨n, hn⟩ := exists_pow_le_of_le_radical_of_fg h hI; ⟨n, le_bot_iff.mp hn⟩⟩
```

Comment:
```suggestion
lemma isNilpotent_iff_le_nilradical {R : Type*} [CommSemiring R] {I : Ideal R} (hI : I.FG) :
    IsNilpotent I ↔ I ≤ nilradical R :=
  ⟨fun ⟨n, hn⟩ _ hx ↦ ⟨n, hn ▸ Ideal.pow_mem_pow hx n⟩,
    fun h ↦ let ⟨n, hn⟩ := exists_pow_le_of_le_radical_of_fg h hI; ⟨n, le_bot_iff.mp hn⟩⟩
```

Reply:
#find_home says this can go into Mathlib.RingTheory.Noetherian.Nilpotent (which would eliminate the large-import), and indeed the only lemma there is a consequence of this. It may be a good idea to rename the file (Finiteness.Nilpotent?) but it's also acceptable as is.